### PR TITLE
LDAP autobind authenticateAsDN for BIND named

### PR DESCRIPTION
**Pull request by tiran**
Use new nsslapd-ldapiautoauthdnattr feature to switch BIND named from
GSSAPI bind to EXTERNAL LDAPI bind.

Requires 389-DS build https://copr.fedorainfracloud.org/coprs/build/1832975

- [x] Requires new build of 389-DS

Fixes: https://pagure.io/freeipa/issue/8544
See: https://github.com/389ds/389-ds-base/issues/4381
Signed-off-by: Christian Heimes <cheimes@redhat.com>

### CODE REVIEW

File: ipatests/prci_definitions/gating.yaml [hunk 1]
```diff
@@ -30,8 +30,8 @@ jobs:
         git_repo: '{git_repo}'
         git_refspec: '{git_refspec}'
         template: &ci-master-latest
-          name: freeipa/ci-master-f32
-          version: 0.0.16
+          name: freeipa/ci-master-f33
+          version: 0.0.4
```
Comment (netoarmando) [inline ipatests/prci_definitions/gating.yaml:34]: @tiran `0.0.5` is available with updated packages.

Comment (tiran) [reply ipatests/prci_definitions/gating.yaml]: Perfect, thanks!

**Comment by tiran** [comment]
Test failures are known issues:

* https://pagure.io/freeipa/issue/8506 Nightly failure in ipa-server-install --uninstall: org.freedesktop.DBus.Error.NoReply
* https://pagure.io/freeipa/issue/8666  Nightly test failure (fed33) in test_simple_replication.py::TestSimpleReplication::test_replica_manage

**Comment by abbra** [comment]
we still want this to merge

**Comment by tiran** [comment]
@rcritten @abbra Please take a look. With Azure on F34 tests are finally passing.

**Comment by abbra** [comment]
@tiran could you please rebase on master so that we can re-run this against 389-ds 2.0.5?

**Comment by abbra** [comment]
LGTM from my side.

**Review by fcami — CHANGES_REQUESTED** [review]
Minor nitpick.

**Review by fcami — COMMENTED** [review]
LGTM. Will approve when the requested design change is done.

**Review by fcami — APPROVED** [review]
ACK for when CI passes.

**Comment by abbra** [comment]
master:

* 16e1cbdc5e2192e47f11df4148069a8b7e2220f7 LDAP autobind authenticateAsDN for BIND named
* 5b8f37f88f8e045288f41ffd8c78eeace74b9557 Add design for LDAPI autobind

**Pull request closed without merging**